File: clients/csharp/src/RateLimiting/RateLimitPolicy.cs
using System.Text.Json.Serialization;

namespace MyMind.RateLimiting;

/// <summary>Quota and window declared by the <c>RateLimit-Policy</c> response header.</summary>
public class RateLimitPolicy
{
    [JsonPropertyName("name")]
    public required string Name { get; init; }

    [JsonPropertyName("q")]
    public required int Quota { get; init; }

    [JsonPropertyName("w")]
    public required int WindowSeconds { get; init; }
}

File: clients/swift/src/RateLimiting/RateLimitPolicy.swift
import Foundation

/// Quota and window declared by the `RateLimit-Policy` response header.
public struct RateLimitPolicy: Sendable, Equatable {
    public let name: String
    public let quota: Int
    public let windowSeconds: Int

    public init(name: String, quota: Int, windowSeconds: Int) {
        self.name = name
        self.quota = quota
        self.windowSeconds = windowSeconds
    }
}
